pkg/reflectutil/reflectutil.go

```diff
@@ -0,0 +1,49 @@
+// This file has been adapted from https://github.com/containerd/nerdctl/blob/v1.0.0/pkg/reflectutil/reflectutil.go
+
+package reflectutil
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func UnknownNonEmptyFields(structOrStructPtr interface{}, knownNames ...string) []string {
+	var unknown []string
+	knownNamesMap := make(map[string]struct{}, len(knownNames))
+	for _, name := range knownNames {
+		knownNamesMap[name] = struct{}{}
+	}
+	origVal := reflect.ValueOf(structOrStructPtr)
+	var val reflect.Value
+	switch kind := origVal.Kind(); kind {
+	case reflect.Ptr:
+		val = origVal.Elem()
+	case reflect.Struct:
+		val = origVal
+	default:
+		panic(fmt.Errorf("expected Ptr or Struct, got %+v", kind))
+	}
+	for i := 0; i < val.NumField(); i++ {
+		iField := val.Field(i)
+		if isEmpty(iField) {
+			continue
+		}
+		iName := val.Type().Field(i).Name
+		if _, ok := knownNamesMap[iName]; !ok {
+			unknown = append(unknown, iName)
+		}
+	}
+	return unknown
+}
+
+func isEmpty(v reflect.Value) bool {
+	// NOTE: IsZero returns false for zero-length map and slice
+	if v.IsZero() {
+		return true
+	}
+	switch v.Kind() {
+	case reflect.Map, reflect.Slice:
+		return v.Len() == 0
+	}
+	return false
```

Comment:
Can we check empty structs like this? https://stackoverflow.com/a/52562726

Reply:
The current code is already able to ignore empty structs. 

**For Eg:** 
In below, Firmware is not returned as unknown field
```
Yaml: {Firmware: {}}
```
The issue is more related to how we do in FillDefaults.
https://github.com/lima-vm/lima/blob/master/pkg/limayaml/defaults.go#L189-L191

From our defaults, we are getting
```
Yaml: {Firmware: {pointer.Bool(false)}}
```

We do similar for Mount SSHFS,NinePS setting default if null

Reply:
`Mount.NineP` can be just ignored without any warning (as long as the `MounType` is verified)

The `FillDefault()` function should leave `Video.Display` and `CPUType` empty when the driver is VZ, and the VZ driver should print warnings when those properties are non-empty.

Reply:
Done as per your suggestion